lib/dynflow/dispatcher/executor_dispatcher.rb

```diff
@@ -1,9 +1,10 @@
 module Dynflow
   module Dispatcher
     class ExecutorDispatcher < Abstract
-      def initialize(world)
+      def initialize(world, semaphore)
         @world        = Type! world, World
         @current_futures = Set.new
+        @semaphore = ::Dynflow::Semaphores::Aggregating.new(:global => semaphore)
```

Comment:
The 'global' key might be a bit misleading, I had a bit different plans with this, we can talk about this more at the office.